### ISSUE TITLE
BaseRecalibrator reports different Counts and Observations compared to GATK3 results

### ISSUE THREAD

**Issue by droazen** [broadinstitute/gatk#1414]
We'd had reports that BaseRecalibrator is sometimes off by exactly 1.0 compared to GATK3 results (never 0.9999 or 1.00001, so not sure this is a precision issue).

**Comment by droazen** [comment]
Relates to https://github.com/broadinstitute/gatk/issues/1413

**Comment by akiezun** [comment]
i'm happy to look at this. do you have a testcase?

**Comment by droazen** [comment]
I believe @davidadamsphd has one. Reassigning to you!

**Comment by akiezun** [comment]
@davidadamsphd please send me a testcase

**Comment by droazen** [comment]
@lbergelson says he also has a way to reproduce this issue

**Comment by lbergelson** [comment]
On further investigation it looks like it's not always off by 1.  A lot of lines are exactly 1 off, but many are off by more. 

The differences are in the `Count` column of `GATKTable:Quantized:Quality quantization map` and in the `Observations` columns of the other tables.  The rest of the columns appear unaffected.


```
./gatk-launch BaseRecalibrator -I src/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam -O gatk4.tbl --knownSites src/test/resources/large/dbsnp_138.b37.20.21.vcf -R src/test/resources/large/human_g1k_v37.20.21.fasta
```

```
java -jar ~/Workspace/GenomeAnalysisTK.jar -T BaseRecalibrator -I src/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam -o gatk3.tbl --knownSites src/test/resources/large/dbsnp_138.b37.20.21.vcf -R src/test/resources/large/human_g1k_v37.20.21.fasta --sort_by_all_columns
```

gatk3 and gatk4 sort order is different so sorting before comparing may be helpful

`vimdiff <( sort gatk3.tbl ) <( sort gatk4.tbl )`

**Comment by akiezun** [comment]
here's a samfile with 1 read that produces an empty bqsr table on gatk4 but works fine on gatk3 (after converting to bam because gatk3 needs a bam, unlike gatk4 which does not care):
```
@HD	VN:1.5	GO:none	SO:coordinate
@SQ	SN:20	LN:63025520
@SQ	SN:21	LN:48129895
@RG	ID:20FUK.5	PL:illumina	PU:20FUKAAXX100202.5	LB:Solexa-18483	SM:NA12878	CN:BI
20FUKAAXX100202:5:64:17296:91431	161	20	10189232	25	101M	=	10189225	93	TCTCTTGACTGTTTTTCTCCTGCATATGGGCCACATTTGTTTGTATGTCTTAATATTTAGTTAAGAACTACACATTTTAAATAATATAATGCGGAGATCGG	ADECCBDFCGFFFFFCCFCFHGFGGGGGGGFEGEGGFGGFGCDFGGGGFHGGHGGGGGGGGGGHGHIEHGEHEHGGGGGHIGGHGGGGIGEE?GFGGGF?F	X0:i:1	X1:i:0	MD:Z:94C0A2T0C0T0	RG:Z:20FUK.5	XG:i:0	AM:i:25	NM:i:5	SM:i:25	XM:i:5	XO:i:0	BQ:Z:@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@D^ZbabbbaZ	MQ:i:25	OQ:Z:HHHHHHHHHHHHHHHGGGGHHHHHHHHHHHHHHHHHHHHHHGGHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHGGHHGHHHHHH	XT:A:U
```

**Comment by droazen** [comment]
Is the read getting filtered out in GATK4?

**Comment by akiezun** [comment]
No

On Wednesday, January 6, 2016, droazen <notifications@github.com> wrote:

> Is the read getting filtered out in GATK4?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/gatk/issues/1414#issuecomment-169480149>
> .
>


-- 
Sent from Gmail Mobile

**Comment by akiezun** [comment]
Investigation so far:
1) In the problem is not that the read is filtered out - both gatk3 and gatk4 examine the read
2) the read is unusual in that the insert size is shorter than the read and thus it gets hardclipped before BQSR (in gatk4 at least)

**Comment by akiezun** [comment]
closed by #1419
